### PR TITLE
Add license file, changelog and manifest

### LINKED ISSUE #47: License file?
Hi!

I don't see the license file anymore either on the package or here in the repository. Is that expected?

I ask because I'm one of the maintainers of this package on [conda-forge](https://github.com/conda-forge/httpcore-feedstock), and we require all packages to have a license.

### PR DESCRIPTION
**Pull request by yeraydiazdiaz**
Closes #47

### CODE REVIEW

**Comment by yeraydiazdiaz** [comment]
> Which license is it, BSD-3-Clause I assume?

Tbh I just [copied it from HTTPX](https://github.com/encode/httpx/blob/master/LICENSE.md) and GitHub seems to think it is BSD-3 🙂 

> We already set package_data in setup.py, do we need to remove it?

Good catch. Looks like they [complement each other](https://packaging.python.org/guides/using-manifest-in/#how-files-are-included-in-an-sdist) so I guess we should pick one or the other to avoid confusion but, again, I was following HTTPX. If I had to choose I guess `package_data` means one less file in the repo? 🤷‍♂️

**Comment by florimondmanca** [comment]
Oh, okay.

Hmm, I’m more used to see a MANIFEST.in than a list of files in setup.py, but as long as both work (ie the files are indeed included in the dist) then we can go for either, I guess.

File: setup.py [hunk 1]
```diff
@@ -50,15 +50,10 @@ def get_packages(package):
...
-    package_data={"httpcore": ["py.typed"]},
     packages=get_packages("httpcore"),
     include_package_data=True,
     zip_safe=False,
-    install_requires=[
-        "h11>=0.8,<0.10",
-        "h2==3.*",
-        "sniffio==1.*",
-    ],
+    install_requires=["h11>=0.8,<0.10", "h2==3.*", "sniffio==1.*",],
```
Comment (florimondmanca) [inline setup.py:56]: Black tried to sneak this one in ;-)

```suggestion
    install_requires=["h11>=0.8,<0.10", "h2==3.*", "sniffio==1.*"],
```

**Comment by yeraydiazdiaz** [comment]
Thanks for reviewind @florimondmanca 🎉

**Comment by tomchristie** [comment]
> Hmm, I’m more used to see a MANIFEST.in than a list of files in setup.py, but as long as both work (ie the files are indeed included in the dist) then we can go for either, I guess.

MANIFEST.in is the way to go here yup (we ought to update other encode projects in line with this too).

There’s at least one existing issue on the Starlette’s repo regarding packaging issues when including extra files using setup.py instead of the manifest file.

(And thanks @yeraydiazdiaz!)

**Pull request merged**